src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/SourceInfo.cs

```diff
@@ -5,11 +5,15 @@
 using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis;
 using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.ValueContentAnalysis;
+using Microsoft.CodeAnalysis.Operations;
 
 namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
 {
-    public delegate bool IsInvocationTaintedWithPointsToAnalysis(IEnumerable<PointsToAbstractValue> argumentPointsTo);
-    public delegate bool IsInvocationTaintedWithValueContentAnalysis(IEnumerable<PointsToAbstractValue> argumentPonitsTos, IEnumerable<ValueContentAbstractValue> argumentValueContents);
+    using PointsToChecksAndTargets = ImmutableDictionary<IsInvocationTaintedWithPointsToAnalysis, ImmutableHashSet<string>>;
+    using ValueContentChecksAndTargets = ImmutableDictionary<IsInvocationTaintedWithValueContentAnalysis, ImmutableHashSet<string>>;
+
+    public delegate bool IsInvocationTaintedWithPointsToAnalysis(IEnumerable<IArgumentOperation> arguments, IEnumerable<PointsToAbstractValue> argumentPointsTo);
```

Comment:
>IEnumerable<IArgumentOperation> arguments [](start = 65, length = 41)

I think the arguments will always be an ImmutableArray<IArgumentOperation>?  I vote we use that as the type, so that accessing and checking the arguments is more natural (like arguments[0].Parameter.SpecialType == SpecialType.System_String).